src/core/render/index.js

```diff
@@ -13,290 +12,302 @@ import { Compiler } from './compiler';
+              raw: result,
+            },
+            tokens => {
+              html = this.compiler.compile(tokens);
+              html = this.isRemoteUrl ? DOMPurify.sanitize(html) : html;
+              callback();
+              next();
+            }
+          );
+        }
+      });
+    }
 
-      if (this.isHTML) {
-        html = this.result = text;
-        callback();
-        next();
-      } else {
-        prerenderEmbed(
-          {
-            compiler: this.compiler,
-            raw: result,
-          },
-          tokens => {
-            html = this.compiler.compile(tokens);
-            html = this.isRemoteUrl ? DOMPurify.sanitize(html) : html;
-            callback();
-            next();
-          }
-        );
+    _renderCover(text, coverOnly) {
+      const el = dom.getNode('.cover');
+
+      dom.toggleClass(
+        dom.getNode('main'),
+        coverOnly ? 'add' : 'remove',
+        'hidden'
+      );
+      if (!text) {
+        dom.toggleClass(el, 'remove', 'show');
+        return;
       }
-    });
-  };
 
-  proto._renderCover = function(text, coverOnly) {
-    const el = dom.getNode('.cover');
-
-    dom.toggleClass(
-      dom.getNode('main'),
-      coverOnly ? 'add' : 'remove',
-      'hidden'
-    );
-    if (!text) {
-      dom.toggleClass(el, 'remove', 'show');
-      return;
-    }
+      dom.toggleClass(el, 'add', 'show');
 
-    dom.toggleClass(el, 'add', 'show');
+      let html = this.coverIsHTML ? text : this.compiler.cover(text);
 
-    let html = this.coverIsHTML ? text : this.compiler.cover(text);
+      const m = html
+        .trim()
+        .match('<p><img.*?data-origin="(.*?)"[^a]+alt="(.*?)">([^<]*?)</p>$');
 
-    const m = html
-      .trim()
-      .match('<p><img.*?data-origin="(.*?)"[^a]+alt="(.*?)">([^<]*?)</p>$');
+      if (m) {
+        if (m[2] === 'color') {
+          el.style.background = m[1] + (m[3] || '');
+        } else {
+          let path = m[1];
 
-    if (m) {
-      if (m[2] === 'color') {
-        el.style.background = m[1] + (m[3] || '');
-      } else {
-        let path = m[1];
+          dom.toggleClass(el, 'add', 'has-mask');
+          if (!isAbsolutePath(m[1])) {
+            path = getPath(this.router.getBasePath(), m[1]);
+          }
 
-        dom.toggleClass(el, 'add', 'has-mask');
-        if (!isAbsolutePath(m[1])) {
-          path = getPath(this.router.getBasePath(), m[1]);
+          el.style.backgroundImage = `url(${path})`;
+          el.style.backgroundSize = 'cover';
+          el.style.backgroundPosition = 'center center';
         }
 
-        el.style.backgroundImage = `url(${path})`;
-        el.style.backgroundSize = 'cover';
-        el.style.backgroundPosition = 'center center';
+        html = html.replace(m[0], '');
       }
 
-      html = html.replace(m[0], '');
+      this._renderTo('.cover-main', html);
+      sticky();
     }
 
-    this._renderTo('.cover-main', html);
-    sticky();
-  };
+    _render_updateRender() {
```

Comment:
There was previously two functions, both named `updateRender`, one in the router mixin, one in the render mixin, which is confusing. They were both plain `functions` before (so basically "private" methods), but now that they are both on `this`, I renamed them to `_render_updateRender` and `_router_updateRender` so they can be differentiated. We can clean it or find better names later...

Reply:
Apart from this change, nothing major other than converting to classes and all functions are methods.